toolchain/check/handle_name.cpp

```diff
@@ -157,10 +177,8 @@ auto HandleQualifiedDeclaration(Context& context, Parse::Node parse_node)
     pop_and_apply_first_child();
     context.declaration_name_stack().ApplyNameQualifier(parse_node2, name_id2);
   } else {
-    SemIR::NodeId node_id2 = context.node_stack().PopExpression();
-    pop_and_apply_first_child();
-    context.declaration_name_stack().ApplyExpressionQualifier(parse_node2,
-                                                              node_id2);
+    CARBON_FATAL() << "Unexpected node kind on right side of qualified "
```

Comment:
The `Pop` on line 176 includes a CHECK. As a consequence, you should be able to remove the `if` and this `FATAL` and get equivalent behavior without running `PeekParseNode`.

Reply:
Yes, thank you :) That's much nicer.